src/kartograf/atom_mapper.py

```diff
@@ -863,28 +862,78 @@ def suggest_mapping_from_rdmols(
 
         return mapping
 
+    @staticmethod
+    def _split_component_molecules(component: Union[SmallMoleculeComponent, ProteinComponent]) -> list[Chem.Mol]:
+        """
+        Aims at splitting a component into its disconected components based on the
+        connectivity of the molecules that compose it. Useful for mapping multimer components
+        or proteins with structural waters or similarly.
+
+        This returns a list of ``Chem.Mol`` objects with a prop named `Starting_index` indicating the starting
+        index in the original component.
+        """
+        from rdkit.Chem.rdmolops import GetMolFrags
+        rdmol = component.to_rdkit()
+        index_tuples = []
+        fragments = GetMolFrags(rdmol, asMols=True, sanitizeFrags=True, fragsMolAtomMapping=index_tuples)
+        for fragment, index_tuple in zip(fragments, index_tuples):
+            fragment.SetIntProp("Starting_index", index_tuple[0])
+        return fragments
+
+
     def suggest_mappings(
-            self, A: SmallMoleculeComponent, B: SmallMoleculeComponent
-    ) -> Iterator[AtomMapping]:
+            self, A: Union[SmallMoleculeComponent, ProteinComponent], B: Union[SmallMoleculeComponent, ProteinComponent]
```

Comment:
How about just typing on ExplicitMoleculeComponent? https://github.com/OpenFreeEnergy/gufe/blob/main/gufe/components/explicitmoleculecomponent.py#L99

Reply:
Yeah we can I wasn't sure if we should be explicit or not, I did also notice that I was changing the type hint from the base class as well not sure if it should be changed in [gufe](https://github.com/OpenFreeEnergy/gufe/blob/d15ea96dd98565d00e94949eb62cff08299b5bd6/gufe/mapping/atom_mapper.py#L22-L24)

Reply:
Probably not worth being too explicit - just in case someone decides to create another Component type (I've been guilty of that in some cases).

Just going for `Component` is also maybe ok, but I kinda agree that it might make sense to switch to `ExplicitMoleculeComponent` upstream.